Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs

```diff
@@ -44,23 +51,30 @@ public TestConductRepository(TrappistDbContext dbContext)
 
         public async Task<TestInstructionsAC> GetTestInstructionsAsync(string testLink)
         {
-            var testDeatilsObj = await _dbContext.Test.Where(x => x.Link == testLink)
-                .Include(x => x.TestQuestion)
-                .Include(x => x.TestCategory)
-                .ThenInclude(x => x.Category).ToListAsync();
-            if (testDeatilsObj != null)
+            var testObject = await _dbContext.Test.Where(x => x.Link == testLink)
+                    .Include(x => x.TestQuestion).ToListAsync();
+            if(testObject.Any())
             {
-                var testInstructionDetails = testDeatilsObj.First();
-                var totalNumberOfQuestions = testInstructionDetails.TestQuestion.Count();
-                var testCategoryNameList = testInstructionDetails.TestCategory.Select(x => x.Category.CategoryName).ToList();
+                var testInstructionsDetails = testObject.First();
+                var totalNumberOfQuestions = testInstructionsDetails.TestQuestion.Count();
+                var testCategoryANameList = new List<string>();
+                var testByIdObj= await _testRepository.GetTestByIdAsync(testInstructionsDetails.Id, testInstructionsDetails.CreatedByUserId);
+                var categoryAcList = testByIdObj.CategoryAcList;
+                foreach(var category in categoryAcList)
+                {
+                    if(category.NumberOfSelectedQuestion!=0 && category.IsSelect!= false)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the unnecessary Boolean literal(s). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1125)